### DISCUSSION TITLE
UCLASS와 TSubclassOf<T>의 차이가 뭐죠?

### DISCUSSION THREAD

**Discussion by simeddk** [simeddk/Tore04_BasicSyntaxCPP#4] [Q&A] [answered]
귀찮은데 UCLASS만 쓰면 안되나요?
이런 변수 하나 만들어놓고 스폰액터에서 써먹을려고 하는데
사수가 눈꺼풀이 뒤집어 지면서 쌍욕을 하네요ㅠㅠ

```c++
<MyClass.h>
UPROPERTY(EditAnywhere)
UCLASS* SomeClassRef;

<MyClass.cpp>
GetWorld()->SpawnActor<AActor>(SomeClassRef, FTransform());
```

대체 뭐가 문제일까요? 그냥 억까겠죠?

**Comment by HyeonLang** [comment, marked as answer]
UCLASS 사용시 에디터에서 class를 넣을 때 불필요한 클래스들까지 노출됩니다.
TSubclassOf<T> 사용시 에디터에서 자신과 자식 클래스만이 들어갈 수 있도록 합니다.

**Comment by ehddnjs1200** [comment]
정동원

UCLASS는 클래스 선언 시 사용되며, 클래스 타입 포인터로 사용할 수 없으며 또한 UCLASS 매크로는 클래스의 메타데이터를 생성할 때 사용되는 것이지 클래스 타입을 가리키는 포인터 타입 아닙니다 따라서 클래스 타입을 변수로 참조할 때는
TSubclassOf<AActor>를 사용하여 원하는 클래스를 안전하게 참조하는 게 좋습니다
